koku/api/report/queries.py

```diff
@@ -181,11 +182,28 @@ def add(self, query_filter=None, table=None, field=None, operation=None, paramet
 
     def compose(self):
         """Compose filters into a dict for submitting to Django's ORM."""
-        out = {}
+        composed_query = None
+        compose_dict = {}
         for filt in self._filters:
-            if filt.parameter is not None:
-                out.update(filt.composed_dict())
-        return out
+            filt_key = filt.composed_query_string()
+            if compose_dict.get(filt_key) is None:
```

Comment:
Definitely no reason to change this now, but I figured I'd insert some annoying Python standard library advice and say the collections package has a defaultdict class that lets you set an automatic default value for the value of a dict key. Set `compose_dict = defauldict(list)` and then you can just append away without checking if the key exists yet.